test/common.py

```diff
@@ -1875,29 +1875,27 @@ class BrowserCore(RunnerCore):
   def __init__(self, *args, **kwargs):
     super().__init__(*args, **kwargs)
 
-  @staticmethod
-  def browser_open(url):
-    if not EMTEST_BROWSER:
-      logger.info('Using default system browser')
-      webbrowser.open_new(url)
-      return
+  @classmethod
+  def browser_restart(cls):
+    # Kill existing browser
+    logger.info("Restarting browser process")
+    cls.browser_proc.terminate()
+    try:
+      cls.browser_proc.wait(1)
+    except subprocess.TimeoutExpired:
+      cls.browser_proc.kill()
+      cls.browser_proc.wait()
```

Comment:
Maybe add a comment about why it makes sense to give it a 1-second chance to exit gracefully? I'm not sure why that is, actually.

Reply:
Before this PR we were using python's `webbrowser.get(..).open_new(url)` which I assume would open the system browser on windows.

After this PR we default to using `subprocess.Popen(['google-chrome'...)` .. which doesn't work unless you have that browser in the PATH. 

However, in this case we don't want to be starting any browser at all.

Reply:
Sorry, I'm confused. I asked about the terminate/wait/kill/wait sequence here, but you replied about starting the browser? Are those connected in a way I do not understand?

Reply:
Somehow the replies to these comments got mixed up.    Here I'm commenting on as to why the EMTEST_BROWSER=0 is needed when running the socket tests on windows.

Reply:
Regarding the using of `terminate()` prior to `kill()` I added comment.

It just seemed sensible to me to give the browser a chance to exit cleanly before killing with fire.